modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java

```diff
@@ -957,6 +957,19 @@ public CPDefinition deleteCPDefinition(CPDefinition cpDefinition)
 			cProductLocalService.deleteCProduct(cpDefinition.getCProductId());
 		}
 
+		if (cpDefinitionsCount > 1) {
+			CProduct cProduct = cProductLocalService.getCProduct(
+				cpDefinition.getCProductId());
+
+			long publishedCPDefinitionId =
+				cProduct.getPublishedCPDefinitionId();
+
+			if (publishedCPDefinitionId == cpDefinition.getCPDefinitionId()) {
+				cProductLocalService.updatePublishedCPDefinitionId(
+					cProduct.getCProductId(), 0);
```

Comment:
Check if there is an old approved version, if there is one, update the CProduct with the old approved CPDefinitionId